lib/hc-kernel-assemble.in

```diff
@@ -88,20 +88,20 @@ if [ -f "$2" ]; then
   mv "$2" "$TEMP_DIR/$BASENAME.tmp.o"
 fi
 
-$LLVM_DIS "$KERNEL_INPUT" -o "$TEMP_NAME.ll"
+cp $KERNEL_INPUT "$TEMP_NAME.bc"
 if [ $KMDUMPLLVM == "1" ]; then
+  $LLVM_DIS "$TEMP_NAME.bc" -o "$TEMP_NAME.ll"
   cp "$TEMP_NAME.ll" ./dump.kernel_input.ll
 fi
-$OPT -load $LIBPATH/LLVMDirectFuncCall.so -redirect < "$TEMP_NAME.ll" 2>"$TEMP_NAME.kernel_redirect.ll" >/dev/null
+$OPT -load $LIBPATH/LLVMDirectFuncCall.so -redirect < "$TEMP_NAME.ll" 2>"$TEMP_NAME.kernel_redirect.bc" >/dev/null
```

Comment:
does the input need to be in llvm ir text?

Reply:
Sorry, it seems that this commit may be outdated, let me fix these issues